### PR TITLE
Remove usage of `qml.Hamiltonian` from `hamiltonian.py`

### PR DESCRIPTION
**Pull request by astralcai**
`hamiltonian.py` defines the legacy `Hamiltonian`. It should not be returning `LinearCombination` in its own methods even with new op math enabled.

### CODE REVIEW

**Comment by astralcai** [comment]
[sc-59501]

**Review by Qottmann — APPROVED** [review]

**Review by lillian542 — APPROVED** [review]
🎉

**Comment by Qottmann** [comment]
same for this, wait for ham-tests CI to pass before merging

**Pull request merged**
